### PR TITLE
Update SafetyNet attestation description

### LINKED ISSUE #968: SafetyNet Attestation Clarifications
Splitting off the SafetyNet conversation from #950, as requested during the last call.

I think the questions are:

1. What is `ver` and how is it verified and / or used during verification?
2. Where do we find the root cert for SafetyNet?
3. What value gets put in the SafetyNet `nonce` (not part of #950, but perhaps some clarification is needed)?

cc: @agl @christiaanbrand @leshi @kpaulh

### LINKED ISSUE #1135: Refer android-safetynet verification to SafetyNet documentation?
@arnar notes in https://github.com/w3c/webauthn/issues/1132#issuecomment-454998300:

> I wonder if it's best to simply refer to [SafetyNet documentation](https://developer.android.com/training/safetynet/attestation#verify-compat-check) on this, rather than chasing any potential changes there.

I believe this has been suggested off-hand before, too. We've had several inaccuracies in the verification steps copied into WebAuthn: #968, #1018, #1132.

Should we delete the [android-safetynet](https://w3c.github.io/webauthn/#android-safetynet-attestation) verification procedure from WebAuthn and just refer to the [SafetyNet documentation](https://developer.android.com/training/safetynet/attestation#verify-compat-check) instead?

### PR DESCRIPTION
**Pull request by leshi**
Use official SafetyNet documentation as a reference rather than trying to keep this text up to date.
Also update links to documentation

Closes #1135, closes #968

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1170.html" title="Last updated on Mar 20, 2019, 5:18 PM UTC (6e50173)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1170/26cf7c6...6e50173.html" title="Last updated on Mar 20, 2019, 5:18 PM UTC (6e50173)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -4017,13 +4017,12 @@ data</dfn> is identified by the OID "1.3.6.1.4.1.11129.2.1.17".
 
 ## Android SafetyNet Attestation Statement Format ## {#android-safetynet-attestation}
 
-When the [=authenticator=] in question is a platform-provided Authenticator on certain Android platforms, the attestation
-statement is based on the [SafetyNet API](https://developer.android.com/training/safetynet/index.html#compat-check-response). In
+When the [=authenticator=] is a platform-provided on certain Android platforms, the attestation
```
Comment (emlun) [inline index.bs]: Missing noun: "a platform-provided on certain"

File: index.bs [hunk 2]
```diff
@@ -4071,13 +4070,11 @@ even if the SafetyNet API is also present.
     as follows:
     - Verify that |attStmt| is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to extract the
         contained fields.
-    - Verify that |response| is a valid SafetyNet response of version |ver|.
+    - Verify that |response| is a valid SafetyNet response of version |ver| 
+        by following steps indidated by the [SafetyNet online documentation](https://developer.android.com/training/safetynet/attestation.html#compat-check-response).
```
Comment (emlun) [inline index.bs]: s/following steps indidated/following the steps indicated/

File: index.bs [hunk 3]
```diff
@@ -4071,13 +4070,11 @@ even if the SafetyNet API is also present.
...
     - Verify that the `nonce` attribute in the payload of |response| is identical to the Base64 encoding of the SHA-256 hash of the concatenation of |authenticatorData| and |clientDataHash|.
-    - Let |x5c| be the value of the `x5c` attribute in the header of |response|, parsed as a sequence of X.509 certificates.
-    - Let |attestationCert| be the first element of |x5c|.
-    - Verify that |attestationCert| is issued to the hostname "attest.android.com" (see
-        [SafetyNet online documentation](https://developer.android.com/training/safetynet/index.html#compat-check-response)).
-    - Verify that the `ctsProfileMatch` attribute in the payload of |response| is [TRUE].
+    - Verify that the SafetyNet response actually came from the SafetyNet service by following stepds in the
```
Comment (emlun) [inline index.bs]: s/following stepds/following the steps/

File: index.bs (hunk 2 above)
Comment (leshi) [reply index.bs]: done

File: index.bs (hunk 3 above)
Comment (leshi) [reply index.bs]: done

File: index.bs (hunk 1 above)
Comment (leshi) [reply index.bs]: done

**Comment by arnar** [comment]
LGTM!

File: index.bs [hunk 4]
```diff
@@ -4071,13 +4070,13 @@ even if the SafetyNet API is also present.
...
+        by following the steps indicated by the [SafetyNet online documentation](https://developer.android.com/training/safetynet/attestation.html#compat-check-response).
+        As of this writing, there is only one format of the SafetyNet response.  This step is only necessary if the SafetyNet
+        response becomes different for different Google Play Services versions.
     - Verify that the `nonce` attribute in the payload of |response| is identical to the Base64 encoding of the SHA-256 hash of the concatenation of |authenticatorData| and |clientDataHash|.
-    - Let |x5c| be the value of the `x5c` attribute in the header of |response|, parsed as a sequence of X.509 certificates.
-    - Let |attestationCert| be the first element of |x5c|.
-    - Verify that |attestationCert| is issued to the hostname "attest.android.com" (see
-        [SafetyNet online documentation](https://developer.android.com/training/safetynet/index.html#compat-check-response)).
-    - Verify that the `ctsProfileMatch` attribute in the payload of |response| is [TRUE].
+    - Verify that the SafetyNet response actually came from the SafetyNet service by following steps in the
```
Comment (emlun) [inline index.bs]: I still think there's a "the" missing in "following steps" here, but I guess it's also fine enough to leave as is.

**Comment by apowers313** [comment]
Friendly request to Android SafetyNet documentation people: can you mention that the root of trust for SafetyNet is https://pki.goog/ so that people that actually do certificate chain validation (hopefully everybody?) knows where to get their certs.

**Comment by apowers313** [comment]
Slight proposed modification:

> As of this writing, there is only one format of the SafetyNet response.

to

> As of this writing, there is only one format of the SafetyNet response and any numeric value is acceptable for |ver|.

Or something like that.

**Comment by leshi** [comment]
@apowers313, thanks for the suggestion, I don't think *any* number will due.  It will be positive and reflect some real version of GooglePlayServices.  Having said that, I think RPs should just ignore it for now, which is why I didn't write those details down.

**Comment by leshi** [comment]
@apowers313, I'll wait for your LGTM to merge.

**Comment by apowers313** [comment]
It never explicitly says "ignore ver". Can we do something like:

> As of this writing, there is only one format of the SafetyNet response and |ver| is reserved for future use.

**Comment by leshi** [comment]
Updated  commentary around |ver| as per your suggestion.

**Pull request merged**